### PR TITLE
Feature/jest testing 81

### LINKED ISSUE #81: Setup Jest automated testing
We need to set up an automated testing framework on the build, preferably something like [Jest](https://facebook.github.io/jest/) (as that's the current trendy one). Even a basic setup would be better than nothing at this stage.

Essential for completion:
- Jest setup
- Basic example of test-scripted component that can be used as template for other team members

Bonus points (but can be descoped in this task):
- Test scripts for each component (may need this one to be split out into a new task if desired)
- Setup snapshot testing
- Educate others on automated testing (opportunity for dev tutorial)

### PR DESCRIPTION
**Pull request by peasandwell**
### Description

Wrote some basic test scripts for the reducers/actions and the ``<TextInput />`` component to get the ball rolling.

**Note that now ``npm test`` will run both linting and unit tests. Use ``npm run lint`` if you want to just run the linting task**

### Issues fixed

#81 

### Checklist

- [x] `npm test` returns no warnings or errors.

### CODE REVIEW

File: src/actions/index.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import * as types from '../constants/actionTypes';
+import * as actions from './index';
+const payload = {
+	transactionHash: '0x20797147bd5b2990df11d3c8fb861eb7314fc74fb37a74c9365367ee5f880aab',
+	timeStamp: '0x5a8edf01',
+	topics: [ '', '0xf5a3fa2065e82a7e53c4782f9f5961259d5629b8', '0x5c3216a6d9fbd7cf2afbe9157ff7e7d1db3d7bed' ],
+	data: '0xe1',
+	address: '0xe3b3651b2987f76a45753b5160504a2ab606716b',
```
Comment (dan1elhughes) [inline src/actions/index.test.js]: Can we change this to fake data

File: src/reducers/slices/transactions.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import env from '../../utils/environment';
+
+// all values are inserted as hexadecimal values
+// which are then converted in the reducer
+const newTransaction = {
+	transactionHash: '0x20797147bd5b2990df11d3c8fb861eb7314fc74fb37a74c9365367ee5f880aab',
+	timeStamp: '0x5a8edf01',
+	topics: [ '', '0xf5a3fa2065e82a7e53c4782f9f5961259d5629b8', '0x5c3216a6d9fbd7cf2afbe9157ff7e7d1db3d7bed' ],
+	data: '0xe1',
+	address: '0xe3b3651b2987f76a45753b5160504a2ab606716b',
```
Comment (dan1elhughes) [inline src/reducers/slices/transactions.test.js]: Here too

File: src/reducers/slices/transactions.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+				symbol: env.REACT_APP_DEFAULT_SYMBOL,
+			},
+		};
+
+		expect(transactionsReducer(initialState, action)).toEqual(expectedState);
+	});
+
+	it('should disregard new transactions with similar transactionHash', () => {
+		const initialState = {
+			'0x20797147bd5b2990df11d3c8fb861eb7314fc74fb37a74c9365367ee5f880aab': {
```
Comment (dan1elhughes) [inline src/reducers/slices/transactions.test.js]: Here too

**Comment by ShaunBulbrook** [comment]
@peasandwell Can you update https://github.com/HeathWallace/ethereum-pos/blob/master/.circleci/config.yml to run your tests?

File: src/reducers/slices/transactions.test.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+				from: '0x9f5961259d5629b8',
+				to: '0x7ff7e7d1db3d7bed',
+				symbol: env.REACT_APP_DEFAULT_SYMBOL,
+			},
+		};
+
+		expect(transactionsReducer(initialState, action)).toEqual(expectedState);
+	});
+
+	it('should disregard new transactions with similar transactionHash', () => {
```
Comment (ShaunBulbrook) [inline src/reducers/slices/transactions.test.js]: Can we change 'similar' for identical.

**Comment by dan1elhughes** [comment]
Can you strip out the Enzyme stuff that we're not using

**Pull request merged**